napari/layers/_markers_layer/view/properties.py

```diff
@@ -91,3 +99,34 @@ def change_ndim(self, state):
             self.layer.n_dimensional = True
         else:
             self.layer.n_dimensional = False
+
+    def _on_n_dim_change(self, event):
+        with self.layer.events.n_dimensional.blocker(self._on_n_dim_change):
```

Comment:
Note that if any other functions are subscribed, the loop will be triggered again. As a preventative measure, I would just use a "general" block instead of one specific to this function:
```suggestion
        with self.layer.events.n_dimensional.blocker():
  ```